testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -171,3 +172,15 @@ def test_analysis_class():
     assert_array_equal(results, ana.results)
 
     assert_raises(ValueError, ana_class, 2)
+
+def test_analysis_class_decorator():
+    # Issue #1511
+    # analysis_class should not raise
+    # a DeprecationWarning
+    with no_deprecated_call():
```

Comment:
I prefer if the context only covers the lines we want to test and not the whole test